Tests/Microsoft.AppCenter.Crashes.Test.Windows/CrashesTest.cs

```diff
@@ -181,5 +182,102 @@ public void ProcessPendingErrorsDoesNotCrashOnFileDeletionFailure()
+            
+            // Stub get/read error files.
+            Mock.Get(ErrorLogHelper.Instance).Setup(instance => instance.InstanceGetLastErrorLogFile()).Returns(mockFile);
+            Mock.Get(ErrorLogHelper.Instance).Setup(instance => instance.InstanceReadErrorLogFile(mockFile)).Returns(expectedManagedErrorLog);
+
+            // Start crashes service in an enabled to initiate the process of getting the error report.
+            Crashes.SetEnabledAsync(true).Wait();
+            Crashes.Instance.OnChannelGroupReady(_mockChannelGroup.Object, string.Empty);
+            var hasCrashedInLastSession = Crashes.HasCrashedInLastSessionAsync().Result;
+            var lastSessionErrorReport = Crashes.GetLastSessionCrashReportAsync().Result;
+
+            // Verify results.
+            Assert.IsNotNull(lastSessionErrorReport);
+            Assert.AreEqual(expectedManagedErrorLog.Id.ToString(), lastSessionErrorReport.Id);
+            Assert.IsTrue(hasCrashedInLastSession);
+        }
+
+        [TestMethod]
+        public void OnChannelGroupReadyGetsLastSessionErrorReportWhenEnabledAndNoCrashOnDisk()
+        {
+            var mockErrorLogHelper = Mock.Of<ErrorLogHelper>();
+            ErrorLogHelper.Instance = mockErrorLogHelper;
+
+            // Stub get/read error files.
+            Mock.Get(ErrorLogHelper.Instance).Setup(instance => instance.InstanceGetLastErrorLogFile()).Returns((File)null);
+
+            // Start crashes service in an enabled to initiate the process of getting the error report.
+            Crashes.SetEnabledAsync(true).Wait();
+            Crashes.Instance.OnChannelGroupReady(_mockChannelGroup.Object, string.Empty);
+            var hasCrashedInLastSession = Crashes.HasCrashedInLastSessionAsync().Result;
+            var lastSessionErrorReport = Crashes.GetLastSessionCrashReportAsync().Result;
+
+            // Verify results.
+            Assert.IsNull(lastSessionErrorReport);
+            Assert.IsFalse(hasCrashedInLastSession);
+        }
+
+        [TestMethod]
+        public void OnChannelGroupReadyGetsLastSessionErrorReportWhenEnabledAndInvalidCrashOnDisk()
+        {
+            var mockErrorLogHelper = Mock.Of<ErrorLogHelper>();
+            ErrorLogHelper.Instance = mockErrorLogHelper;
+
+            // Stub get/read error files.
+            Mock.Get(ErrorLogHelper.Instance).Setup(instance => instance.InstanceGetLastErrorLogFile()).Returns(new File());
+            Mock.Get(ErrorLogHelper.Instance).Setup(instance => instance.InstanceReadErrorLogFile(It.IsAny<File>())).Returns((ManagedErrorLog)null);
+
+            // Start crashes service in an enabled to initiate the process of getting the error report.
+            Crashes.SetEnabledAsync(true).Wait();
+            Crashes.Instance.OnChannelGroupReady(_mockChannelGroup.Object, string.Empty);
+            var hasCrashedInLastSession = Crashes.HasCrashedInLastSessionAsync().Result;
```

Comment:
Don't you need to wait until Crashes service processes errors?

Reply:
There's no need. In fact, to be consistent with Android implementation, we shouldn't wait.